lib/compat/wordpress-6.3/html-api/class-gutenberg-html-tag-processor-6-3.php

```diff
@@ -1442,20 +1443,22 @@ private function apply_attributes_updates() {
 			$tail_delta = 0;
 
 			foreach ( $this->lexical_updates as $diff ) {
-				$update_head = $bookmark->start >= $diff->start;
-				$update_tail = $bookmark->end >= $diff->start;
-
-				if ( ! $update_head && ! $update_tail ) {
+				if ( ! $bookmark->start >= $diff->start && ! $bookmark->end >= $diff->start ) {
 					break;
 				}
 
+				if ( $bookmark->start >= $diff->start && $bookmark->end < $diff->end ) {
```

Comment:
this now reads better to me. I think the symmetry of start/end and the simplicity of bookmark/diff helps us out